### ISSUE TITLE
[bug] search doesn't work with no default home

### ISSUE THREAD

**Issue by cridam** [kubernetes/helm#2837]
mv $HOME/.helm /charts
echo $HELM_HOME
/charts/.helm

helm search stable
**Error: open /.helm/repository/repositories.yaml: no such file or directory**

helm search stable --home /charts/.helm
**Error: open /.helm/repository/repositories.yaml: no such file or directory**

helm repo list
stable    	https://kubernetes-charts.storage.googleapis.com

helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈ Happy Helming!⎈ 

helm version 
Client: &version.Version{SemVer:"v2.5.0", GitCommit:"012cb0ac1a1b2f888144ef5a67b8dab6c2d45be6", GitTreeState:"clean"}

**Comment by bacongobbler** [comment]
I was unable to reproduce the root issue in v2.6.0:

```
><> mv ~/.helm/ /tmp/helm
><> export HELM_HOME=/tmp/helm/
><> helm search stable
NAME                         	VERSION	DESCRIPTION                                       
stable/acs-engine-autoscaler 	1.0.0  	Scales worker nodes within agent pools
...
```

However, I was able to reproduce after I ran `helm repo update`, because it reads $HOME.

@cridam I would suggest upgrading to v2.5.1 first and see if that fixes the majority of your issues. I'll see what I can do about `helm repo update`. There were a number of $HOME-related fixes in v2.5.1 that should solve your issue.

**Comment by bacongobbler** [comment]
I found that my issue occurs because the cache path in `$HELM_HOME/repository/repositories.yaml` contains absolute paths to cache indexes, which is loaded during [`repo.LoadRepositoriesFile()`](https://github.com/kubernetes/helm/blob/a2323f8658b29dedc8485020730556793f5d14e0/pkg/repo/repo.go#L56-L66).

```
><> cat /tmp/helm/repository/repositories.yaml 
apiVersion: v1
generated: 2017-05-12T17:12:36.856307157-07:00
repositories:
- caFile: ""
  cache: /Users/bacongobbler/.helm/repository/cache/stable-index.yaml
  certFile: ""
  keyFile: ""
  name: stable
  url: https://kubernetes-charts.storage.googleapis.com
- caFile: ""
  cache: /Users/bacongobbler/.helm/repository/cache/deis-index.yaml
  certFile: ""
  keyFile: ""
  name: deis
  url: https://charts.deis.com/workflow
- caFile: ""
  cache: /Users/bacongobbler/.helm/repository/cache/incubator-index.yaml
  certFile: ""
  keyFile: ""
  name: incubator
  url: https://kubernetes-charts-incubator.storage.googleapis.com/
```

This file is written out during `helm init`, which is why when I `mv $HELM_HOME` to another directory, some repo commands may fail.

Either way this is unrelated to OPs issue. @cridam I still suggest you should update to v2.5.1 and see if that fixes things.

**Comment by cridam** [comment]
I confirm issues are fixed using 2.6.0 version.
Thanks !

**Comment by bacongobbler** [comment]
following up with my issue in #2839.
